scripts/lib/mkcloud-ipmi.sh

```diff
@@ -0,0 +1,45 @@
+function ipmi_cleanup
+{
+    for node in $(seq $nodenumber) ; do
+        if ip link show v${cloud}${node}-x >/dev/null ; then
+            ip link del dev v${cloud}${node}-x
+        fi
+        screen -X -S $cloud-node$node-bmc-lan quit
+    done
+}
+
+function mkveth
+{
+    local nodenumber=$1
+    local bmc_addr=$2
+    local iface=v$cloud$nodenumber
+    ip link add dev $iface-x type veth peer name $iface-y
+    ip link set dev $iface-x up
+    ip link set dev $iface-y up
+    ip addr add ${bmc_addr}/${adminnetmask} dev $iface-x
+    brctl addif $cloudbr $iface-y
+}
+
+function generate_lan_config()
+{
+    local nodenumber=$1
+    local user=$2
+    local password=$3
+    local listenaddr=$4
+    local nodename=$cloud-node$nodenumber
+    local lanconfig=/tmp/ipmi-lan-$nodename.conf
```

Comment:
@dirkmueller we already hardcode /tmp for libvirt vm definition files (e.g. https://github.com/SUSE-Cloud/automation/blob/b8dca9ad65a4b175cbf345d0995a85c6c82141b8/scripts/lib/mkcloud-libvirt.sh#L409) which are sort of analogous to this, so I thought it was appropriate here. I could use /var/run/mkcloud. It looks like the only thing using that directory is dirmaint, which I have no familiarity with, and I didn't want to trailblaze here. But if that's a better path forward then I can go with that.

@nicolasbock I didn't see the value of using mktemp here, I think it's better for the file name to be consistent. mktemp also defaults to putting files in /tmp so we'd still have to decide a better place for it.